pkg/kubelet/cm/dra/plugin/registration.go

```diff
@@ -179,9 +209,19 @@ func (h *RegistrationHandler) RegisterPlugin(pluginName string, endpoint string,
 
 	// Storing endpoint of newly registered DRA Plugin into the map, where plugin name will be the key
 	// all other DRA components will be able to get the actual socket of DRA plugins by its name.
+	if err := draPlugins.add(pluginInstance); err != nil {
+		cancel(err)
+		// No wrapping, the error already contains details.
+		return err
+	}
 
-	if oldPlugin, replaced := draPlugins.add(pluginInstance); replaced {
-		logger.V(1).Info("DRA plugin already registered, the old plugin was replaced and will be forgotten by the kubelet till the next kubelet restart", "oldEndpoint", oldPlugin.endpoint)
+	// Now cancel any pending ResourceSlice wiping for this plugin.
+	// Only needs to be done once.
+	h.mutex.Lock()
+	defer h.mutex.Unlock()
+	if cancel := h.pendingWiping[pluginName]; cancel != nil {
+		(*cancel)(errors.New("new plugin instance registered"))
+		h.pendingWiping[pluginName] = nil
```

Comment:
I noticed it looks like we never call `delete()` on the `pendingWiping` map, so the map might grow forever even if it's filled with `nil` values. As long as the second `ok`/`exists` value of a map lookup isn't used, would this be equivalent to here and the other place where the value in the map is set to `nil`?

```suggestion
		delete(h.pendingWiping, pluginName)
```

I can't imagine that would cause any real problems by itself unless millions of different DRA plugins are registered over the lifetime of this kubelet though.

Reply:
I need to look into the details, but it is indeed better to delete obsolete entries.